File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int t) {
        int n=nums.size();
        unordered_map<int,int> map;
        for(int i=0;i<n;i++){
            
            if(map.find(t-nums[i])!=map.end()){
                return {map[t-nums[i]],i};
            }
            map[nums[i]]=i;
        }
        return {};
    }
};

File: 0001-two-sum/0001-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, t) {
    const map1=new Map();
    for(let i=0;i<nums.length;i++){
        const v=t-nums[i];
        if(map1.has(v)){
            return [map1.get(v),i];
        }
        // map1[nums[i]]=i;
        map1.set(nums[i],i);
    }
    return [-1,-1];
};